idea/data-storage/src/gear/gear.service.ts

```diff
@@ -5,36 +5,42 @@ import { filterEvents } from '@polkadot/api/util';
 import { GenericEventData } from '@polkadot/types';
 import { ExtrinsicStatus } from '@polkadot/types/interfaces';
 import { SignedBlockExtended } from '@polkadot/api-derive/types';
-import { Keys } from '@gear-js/common';
+import { Keys, CodeStatus } from '@gear-js/common';
 import { plainToInstance } from 'class-transformer';
 import { VoidFn } from '@polkadot/api/types';
+import { Option } from '@polkadot/types';
 
 import { ProgramService } from '../program/program.service';
 import { MessageService } from '../message/message.service';
 import { CodeService } from '../code/code.service';
-import { getExtrinsics, getMetaHash, getPayloadAndValue, getPayloadByGearEvent } from '../common/helpers';
+import { getExtrinsics, getMetahash, getPayloadAndValue, eventDataHandlers } from '../common/helpers';
+import { MessageEntryPoint, MessageType } from '../common/enums';
 import { Code, Message, Program } from '../database/entities';
-import { CodeStatus, MessageEntryPoint, MessageType } from '../common/enums';
 import { CodeRepo } from '../code/code.repo';
-import { CodeChangedInput } from '../code/types';
 import { changeStatus } from '../healthcheck/healthcheck.controller';
 import { ProgramRepo } from '../program/program.repo';
-import configuration from '../config/configuration';
 import { BlockService } from '../block/block.service';
 import { RabbitmqService } from '../rabbitmq/rabbitmq.service';
 import { MetaService } from '../meta/meta.service';
+import { sleep } from '../utils/sleep';
+import config from '../config/configuration';
 
-const { gear } = configuration();
+const gearConfig = config().gear;
 
 const MAX_RECONNECTIONS = 10;
 let reconnectionsCounter = 0;
-
+interface BlockParams {
```

Comment:
Please create a `types` folder in `gear` folder to contain the types of this service